### ISSUE TITLE
Latest nightly build (15.7.20180704.1) crashes Visual Studio

### ISSUE THREAD

**Issue by TeaDrivenDev** [Microsoft/visualfsharp#5305]
The Visual F# Tools nightly build version 15.7.20180704.1 causes Visual Studio to close and restart when building or cleaning a solution that includes F# projects.

#### Repro steps

With 15.7.20180704.1 installed, clean, build or debug a solution containing at least one F# project.

#### Expected behavior

After the operation completes, Visual Studio remains usable. If debugging, the application starts and the debugger is attached.

#### Actual behavior

Upon processing the last(?) (F#) project, Visual Studio hangs for a number of seconds, then closes without any error message, restarts and reopens the solution. This does not occur for solutions that do not contain any F# projects.

At least on small solutions, the original operation apparently completes before Visual Studio restarts. If that was "clean", a subsequent cleaning finds nothing to do and completes without impacting Visual Studio. If it was "debug", the assemblies correctly compiled, and the application is started and the debugger attached.

#### Known workarounds

Fall back to 15.7.20180703.1, which does not exhibit the described behavior (and disable automatic updates).

#### Related information 

I have observed the behavior on three different machines, all running Windows 10 with the latest updates and Visual Studio (Professional/Community) 15.7.4. The fix using 15.7.20180703.1 has been validated on two of the machines (I have not tried the third yet).

**Comment by reijerh** [comment]
Same here with 15.7.20180718.1. Here's a stack trace from the Windows event log:
```
Exception Info: System.NullReferenceException
   at Microsoft.VisualStudio.FSharp.ProjectSystem.UIThread.Run(System.Action)
   at Microsoft.VisualStudio.FSharp.ProjectSystem.ProjectNode+<>c__DisplayClass283_1.<DoMSBuildSubmission>b__0(Microsoft.Build.Execution.BuildSubmission)
   at Microsoft.Build.Execution.BuildSubmission.<CheckForCompletion>b__39_0(System.Object)
   at Microsoft.Build.Shared.ThreadPoolExtensions+<>c__DisplayClass0_0.<QueueThreadPoolWorkItemWithCulture>b__0(System.Object)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```
Can indeed reproduce by building/cleaning an F# project.

**Comment by cartermp** [comment]
This should be fixed by #5364, where the `UIThread.Run` call in the legacy project system has been updated to include a fix in the 15.8 branch
